rust/lance/src/index/vector.rs

```diff
@@ -313,6 +314,24 @@ pub(crate) async fn open_vector_index(
                         location: location!(),
                     });
                 }
+
+                let data = FixedSizeListArray::try_from(ivf_pb.centroids_tensor.as_ref().unwrap())?;
+                // train a PQ model on the IVF
```

Comment:
Wait this is training on demand?
So it adds some seconds to cold start, right?

Reply:
yes, but this avoids having to deal with storage format. Will guard this behind a flag